libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/audioguidance/CarAudioGuidanceAction.kt

```diff
@@ -18,26 +17,12 @@ import kotlinx.coroutines.launch
 /**
  * This class creates an action that can control audio guidance.
  */
-class CarAudioGuidanceUi : MapboxActionProvider {
+class CarAudioGuidanceAction {
+
     /**
-     * Android auto action for enabling and disabling the car navigation.
-     * Attach this to the screen while navigating.
+     * Build the [Action].
      */
-    private fun buildSoundButtonAction(screen: Screen): Action {
-        val audioGuidance = MapboxAudioGuidance.getRegisteredInstance()
-        val state = audioGuidance.stateFlow().value
-        return if (!state.isMuted) {
-            buildIconAction(screen, R.drawable.mapbox_car_ic_volume_on) {
-                audioGuidance.mute()
-            }
-        } else {
-            buildIconAction(screen, R.drawable.mapbox_car_ic_volume_off) {
-                audioGuidance.unmute()
-            }
-        }
-    }
-
-    override fun getAction(screen: Screen): Action {
+    fun getAction(screen: Screen): Action {
         screen.lifecycle.apply {
             coroutineScope.launch {
                 repeatOnLifecycle(Lifecycle.State.STARTED) {
```

Comment:
I know it is not related to this PR, but I don't think it is the correct place to do this. `getAction` may be called several times and looks like each time we will launch a new coroutine without canceling the previous one.

Reply:
Added some logs just to watch this thing blow up while fixing. I think we will just make it so 1 audio guidance action is allowed at a time

``` kotlin
    fun getAction(screen: Screen): Action {
        val actionId = UUID.randomUUID().toString()
        logI("kyle_debug") { "getAction $actionId" }
        screen.lifecycle.apply {
            coroutineScope.launch {
                repeatOnLifecycle(Lifecycle.State.STARTED) {
                    MapboxAudioGuidance.getRegisteredInstance().stateFlow()
                        .distinctUntilChanged { old, new ->
                            old.isMuted == new.isMuted && old.isPlayable == new.isPlayable
                        }
                        .collect {
                            logI("kyle_debug") { "getAction collect $actionId $it" }
                            screen.invalidate()
                        }
                }
            }
        }

        return buildSoundButtonAction(screen)
    }
```

```
2022-12-08 13:40:54.673  getAction 61bcfc4a-ce04-46d6-961f-f6d00cc1a363
2022-12-08 13:40:54.674  getAction collect 61bcfc4a-ce04-46d6-961f-f6d00cc1a363 MapboxAudioGuidanceState(isPlayable=true, isMuted=true, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:55.444  getAction d5966233-8455-4277-a93d-f356260d7dc9
2022-12-08 13:40:55.445  getAction collect d5966233-8455-4277-a93d-f356260d7dc9 MapboxAudioGuidanceState(isPlayable=true, isMuted=true, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:57.448  getAction fdcb1894-0611-4fd3-a936-b9c3798db79e
2022-12-08 13:40:57.449  getAction collect fdcb1894-0611-4fd3-a936-b9c3798db79e MapboxAudioGuidanceState(isPlayable=true, isMuted=true, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.453  getAction 61599b79-be61-409a-b16c-3523449ebc9d
2022-12-08 13:40:58.454  getAction collect 61599b79-be61-409a-b16c-3523449ebc9d MapboxAudioGuidanceState(isPlayable=true, isMuted=true, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.654  getAction collect 699276b9-9a91-473d-9996-a8f5653ebc09 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.655  getAction collect f98dedf9-8bf0-41e2-b964-0c8f8d625ad1 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.656  getAction collect d93e5252-e1b2-4b5c-ae31-1c2b2ef5b1ca MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.656  getAction collect e15a8ea3-498c-40b8-850a-1e9757df5a5e MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.657  getAction collect b04acf0b-39cf-4adf-84c3-23a9b05c96b5 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.657  getAction collect b00b2036-3cd2-4d72-bcb1-713203eac19d MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.658  getAction collect 97b7db20-4afa-4535-adc2-9301d8892ada MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.658  getAction collect f5447d30-07f5-4dd2-b31b-94311f338385 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.658  getAction collect 02eac377-9d12-4471-a3f4-3929245cb0af MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.660  getAction collect 1ce05af0-8645-4a15-a574-63f38fa7ee7f MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.660  getAction collect 0b17bd33-8fb2-4385-b65b-bf8da7435309 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.661  getAction collect c51e79d3-256c-4f14-be49-3b17d1f8c95b MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.663  getAction collect 9f7266e5-75e8-4216-9b65-d8026fadcbb5 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.663  getAction collect 0174489f-a338-4331-9acc-a3856230e345 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.664  getAction collect f7746fec-2670-4e4c-a3ca-d464586a45f9 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.664  getAction collect 962cdd7a-a598-4944-b01a-1edfecb0cb1d MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.664  getAction collect d629b49e-fb16-48ed-9423-2eff2c92f2fb MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.665  getAction collect 814f4479-d4da-42e6-8d68-be894894d543 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.665  getAction collect 6396baed-0595-421f-9e85-c6105ab7623a MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.665  getAction collect 6a78deb1-5116-44db-98ab-f6908dc17e99 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.665  getAction collect 61bcfc4a-ce04-46d6-961f-f6d00cc1a363 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.666  getAction collect d5966233-8455-4277-a93d-f356260d7dc9 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.666  getAction collect fdcb1894-0611-4fd3-a936-b9c3798db79e MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 13:40:58.666  getAction collect 61599b79-be61-409a-b16c-3523449ebc9d MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.9875259399414, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
```

Reply:
Ok the logic to do this is not the most elegant but it's fixed. Here are the logs, see that when `collect` is called there is only one subscription.

```
2022-12-08 14:13:31.619 [MapboxAndroidAuto] 2: ActiveGuidanceScreen constructor
2022-12-08 14:13:31.642 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onResume
2022-12-08 14:13:31.658 [MapboxAndroidAuto] 2: ActiveGuidanceScreen loaded
2022-12-08 14:13:31.771 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:31.772 ActiveGuidanceScreen getAction f78f95b2-4b4d-4e7d-b672-120634082a44
2022-12-08 14:13:32.242 ActiveGuidanceScreen getAction collect f78f95b2-4b4d-4e7d-b672-120634082a44 MapboxAudioGuidanceState(isPlayable=true, isMuted=true, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.23226928710938, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 14:13:32.252 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:32.252 ActiveGuidanceScreen getAction bc8245d5-c614-4d4e-9238-d8b90d7383d4
2022-12-08 14:13:32.488 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:32.488 ActiveGuidanceScreen getAction 3d8ca2ab-a76b-48d4-b29a-7c45e57f0f1d
2022-12-08 14:13:32.712 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:32.712 ActiveGuidanceScreen getAction 6f8ecdfe-2ad0-4db2-9f86-a9124f844492
2022-12-08 14:13:33.406 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:33.406 ActiveGuidanceScreen getAction 4983da9f-053b-4154-bfaf-17f80e35d6d9
2022-12-08 14:13:34.407 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:34.407 ActiveGuidanceScreen getAction edb00ca8-a794-4054-a8f2-9164cade40ca
2022-12-08 14:13:34.477 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:34.478 ActiveGuidanceScreen getAction 2e58001b-03c0-4948-9668-e46b020d994a
2022-12-08 14:13:36.508 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:36.508 ActiveGuidanceScreen getAction 9a8fdece-36dd-433d-b440-bd36125df4d0
2022-12-08 14:13:36.852 ActiveGuidanceScreen getAction collect 9a8fdece-36dd-433d-b440-bd36125df4d0 MapboxAudioGuidanceState(isPlayable=true, isMuted=false, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.23226928710938, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 14:13:36.859 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:36.859 ActiveGuidanceScreen getAction f0e9d614-e12f-47e1-9588-47299205773f
2022-12-08 14:13:37.488 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:37.489 ActiveGuidanceScreen getAction 4be283fd-ba50-4a11-92b6-1da8a7c97006
2022-12-08 14:13:38.189 ActiveGuidanceScreen getAction collect 4be283fd-ba50-4a11-92b6-1da8a7c97006 MapboxAudioGuidanceState(isPlayable=true, isMuted=true, voiceInstructions=VoiceInstructions{unrecognized=null, distanceAlongGeometry=109.23226928710938, announcement=Drive southwest on Jefferson Street. Then Turn right onto 12th Street., ssmlAnnouncement=<speak><amazon:effect name="drc"><prosody rate="1.08">Drive southwest on <say-as interpret-as="address">Jefferson Street</say-as>. Then Turn right onto <say-as interpret-as="address">12th Street</say-as>.</prosody></amazon:effect></speak>}, speechAnnouncement=null)
2022-12-08 14:13:38.192 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
2022-12-08 14:13:38.193 ActiveGuidanceScreen getAction 1a544614-8f6d-407e-8b04-45e8c8e57b71
2022-12-08 14:13:38.487 [MapboxAndroidAuto] 2: ActiveGuidanceScreen onGetTemplate
```

Reply:
Thanks for catching! I'm pretty sure that is a big performance issue, please take a look at the fix

Reply:
linking to the logic change https://github.com/mapbox/mapbox-navigation-android/pull/6681#discussion_r1043885596